sys/random_seed/random_seed.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_random
+ *
+ * @{
+ *
+ * @file
+ * @brief       Generate a "unique" seed for PRNGs
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ */
+
+#include <stdio.h>
+
+#include "random_seed.h"
+#include "luid.h"
+#include "xtimer.h"
+#include "periph/adc.h"
+#include "periph/hwrng.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+volatile uint32_t global_random_seed = -1;
+
+uint16_t _seed_by_adc(void)
+{
+    uint16_t val = 0;
+
+    /* init ADC */
+    if (adc_init(ADC_LINE(RANDOM_SEED_ADC_LINE)) != 0) {
+        DEBUG("random_prng_seed: unable to initialize ADC_LINE(%i)\n",
+                RANDOM_SEED_ADC_LINE);
+        return -1;
+    }
+
+    /* use LSB of each ADC sample and shift 16
+        of these LSBs to form an 8-bit integer */
```

Comment:
hmm, 16 shift operations to form a 8-bit integer? I think you mean 16-bit integer?